key/key.js

```diff
@@ -11,6 +12,41 @@ function KeyObservable(root, keyPath) {
 	SettableObservable.call(this, function() {
 		return canKey.get(root, keyPath);
 	}, root);
+
+	var keyPathParts = keyPath.split(".");
+
+	// Determine the last object in the keyPath that this observable actually changes
+	var keyPathSplit = keyPathParts.slice();
+	var lastPartOfPath = keyPathSplit.pop();
+	var restOfPathJoined = keyPathSplit.join(".");
+	var rootMutationObject = (restOfPathJoined) ? canKey.get(root, restOfPathJoined) : root;
+
+	canReflect.assignSymbols(this, {
+		"can.getName": function getName() {
+			var objectName = canReflect.getName(root);
+			return "KeyObservable<" + objectName + "." + keyPath + ">";
+		},
+
+		// Register what this observable changes
+		"can.getWhatIChange": function getWhatIChange() {
+			return {
+				mutate: {
+					keyDependencies: new Map([
+						[rootMutationObject, new Set([lastPartOfPath])]
+					])
+				}
+			};
+		}
+	});
+
+	// Register that this observable changes the objects along the keyPath
+	var i, key, mutationObject;
```

Comment:
what happens when the keys change?  This seems like it only works on initialization.  It probably needs to be able to change.

Reply:
You’re right that this doesn’t work correctly when, say, `outer.inner` changes to a new map.

I’m not sure how best to handle this… should I set up some sort of listener on these objects? Should it only happen when they’re bound? Would that mean I would destroy the listeners when unbound?